app/controllers/service_providers_controller.rb

```diff
@@ -140,6 +143,28 @@ def service_provider_params
   end
   # rubocop:enable MethodLength
 
+  def attach_cert
+    return if params.dig(:service_provider, :cert).blank?
+
+    x509 = OpenSSL::X509::Certificate.new(params[:service_provider].delete(:cert).read)
+
+    service_provider.certs ||= []
+    service_provider.certs << x509.to_pem
```

Comment:
Do we care to do any duplicate prevention?

Reply:
Um, I don't feel too strongly. I guess my follow-up question would be, what kind of duplicate? Do you mean literal duplicate strings? Or duplicate PEM (ex line wrapping etc etc normalized)? Or duplicate attributes ex serial + issuer?

Reply:
I'd only thought to mention it because I was able to upload the same cert file multiple times. It had some strange effect on how the "Remove" input worked, since we'd create them with identical IDs. But I suppose this wouldn't be likely to occur much in practice, and the effects weren't overly problematic anyways.